src/entity.cpp

```diff
@@ -2576,78 +2578,108 @@ void Entity::handleEffects(Stat* myStats)
-			if ( myStats->HP <= 0 )
+			this->flags[BURNING] = false;
+			messagePlayer(player, language[647]); // "The flames go out."
+			serverUpdateEntityFlag(this, BURNING);
+		}
+		else
+		{
+			// If 1 second has passed (50 ticks), process the Burning Status Effect
+			if ( (this->char_fire % TICKS_PER_SECOND) == 0 )
 			{
-				Entity* killer = uidToEntity(myStats->poisonKiller);
-				if ( killer )
+				// Check to see if the fire is put out
+				if ( (rand() % this->chanceToPutOutFire) == 0 )
 				{
-					killer->awardXP(this, true, true);
+					this->flags[BURNING] = false;
+					messagePlayer(player, language[647]); // "The flames go out."
+					serverUpdateEntityFlag(this, BURNING);
 				}
-			}
-			this->setObituary(language[1533]);
-			messagePlayer(player, language[644]);
-			playSoundEntity(this, 28, 64);
-			if ( player == clientnum )
-			{
-				camera_shakey += 3;
-			}
-			else if ( player > 0 && multiplayer == SERVER )
-			{
-				strcpy((char*)net_packet->data, "SHAK");
-				net_packet->data[4] = 0; // turns into 0
-				net_packet->data[5] = 3;
-				net_packet->address.host = net_clients[player - 1].host;
-				net_packet->address.port = net_clients[player - 1].port;
-				net_packet->len = 6;
-				sendPacketSafe(net_sock, -1, net_packet, player - 1);
-			}
-			if ( rand() % 10 == 0 )
-			{
-				if ( myStats->cloak != NULL )
+				else
 				{
-					if ( player == clientnum )
+					this->modHP(-2 - rand() % 3); // Deal between 2 to 5 damage
+
+					// If the Entity died, handle experience
+					if ( myStats->HP <= 0 )
 					{
-						if ( myStats->cloak->count > 1 )
+						Entity* killer = uidToEntity(myStats->poisonKiller);
+						if ( killer != nullptr )
 						{
-							newItem(myStats->cloak->type, myStats->cloak->status, myStats->cloak->beatitude, myStats->cloak->count - 1, myStats->cloak->appearance, myStats->cloak->identified, &myStats->inventory);
+							killer->awardXP(this, true, true);
 						}
 					}
-					myStats->cloak->count = 1;
-					myStats->cloak->status = static_cast<Status>(myStats->cloak->status - 1);
-					if ( myStats->cloak->status != BROKEN )
-					{
-						messagePlayer(player, language[645], myStats->cloak->getName());
-					}
-					else
+
+					// Give the Player feedback on being hurt
+					this->setObituary(language[1533]); // "burns to a crisp."
+					messagePlayer(player, language[644]); // "It burns! It burns!"
+					playSoundEntity(this, 28, 64); // "Damage.ogg"
+
+					// Shake the Camera
+					if ( player == clientnum )
 					{
-						messagePlayer(player, language[646], myStats->cloak->getName());
+						camera_shakey += 3;
 					}
-					if ( player > 0 && multiplayer == SERVER )
+					else if ( player > 0 && multiplayer == SERVER )
 					{
-						strcpy((char*)net_packet->data, "ARMR");
-						net_packet->data[4] = 6;
-						net_packet->data[5] = myStats->cloak->status;
+						strcpy((char*)net_packet->data, "SHAK");
+						net_packet->data[4] = 0; // turns into 0
+						net_packet->data[5] = 3;
```

Comment:
I liked setting the shake to 5 rather than 3 since 3 is very minor shaking for such a deadly effect.